tests/proofs/dexter/dexter-spec.md

```diff
@@ -122,6 +123,39 @@ If any of the conditions are not satisfied, the call fails.
       orBool IsUpdating
       orBool Sender =/=K CurrentManager
       orBool LQTAddress =/=K #Address("tz1Ke2h7sDdakHJQh8WX4Z372du1KChsksyU")
+      */
+```
+
+## Update Token Pool
+
+The contract queries its underlying token contract for its own token balance if the following conditions are satisfied:
+
+1.  the token pool is _not_ currently updating (i.e. `storage.selfIsUpdatingTokenPool = false`)
+2.  exactly 0 tez was transferred to this contract when it was invoked
+3.  the txn sender must be equal to the txn source
+4.  if we are running the FA2 version of Dexter, then check that the contract at address `storage.tokenAddress` has a well-typed FA2 `balance_of` entrypoint;
+    otherwise, check that the contract at address `storage.tokenAddress` has a well-typed FA12 `get_balance` entrypoint.
+
+```k
+  claim <k> #runProof(IsFA2, UpdateTokenPool) => . </k>
+        <stack> .Stack </stack>
+        <selfIsUpdatingTokenPool> false => true </selfIsUpdatingTokenPool>
+        <tokenAddress> TokenAddress:Address </tokenAddress>
+
+        <myaddr> SelfAddress </myaddr>
+        <myamount> #Mutez(Amount) </myamount>
+        <senderaddr> Sender </senderaddr>
+        <sourceaddr> Sender </sourceaddr>
+        <paramtype> #Type(nat) </paramtype>  // Why does this work????? Why would this be the type???
+        <knownaddrs> KnownAddresses </knownaddrs>
+        <returncode> 111 => ?_ </returncode> //  This is necessary if we hit an #Assume(false), I think that means we're on an "impossible" branch? So it's correct to include it?
+        <operations> _ => [ Transfer_tokens Pair SelfAddress #Contract ( SelfAddress , nat ) #Mutez (0) TokenAddress ?_ ] ;; .InternalList </operations>
+        <nonce> #Nonce(O) => #Nonce(O +Int 1) </nonce>
+    requires Amount ==Int 0
+     andBool notBool IsFA2
+     andBool TokenAddress in_keys(KnownAddresses)
+     andBool KnownAddresses[TokenAddress] ==K #Contract(TokenAddress, #Type(pair address (contract nat)))
+     andBool KnownAddresses[SelfAddress] ==K #Contract(SelfAddress, SelfType)
```

Comment:
```suggestion
        <paramtype> #Type(nat) </paramtype>  // NOTE: return type of `update_token_pool_internal` entrypoint
        <knownaddrs> KnownAddresses </knownaddrs>
        <operations> _ => [ Transfer_tokens Pair SelfAddress #Contract ( SelfAddress , nat ) #Mutez(0) TokenAddress ?_ ] ;; .InternalList </operations>
        <nonce> #Nonce(O) => #Nonce(O +Int 1) </nonce>
    requires Amount ==Int 0
     andBool notBool IsFA2
     andBool TokenAddress in_keys(KnownAddresses)
     andBool KnownAddresses[TokenAddress] ==K #Contract(TokenAddress, #Type(pair address (contract nat)))
```

Reply:
The `<returncode>` cell modification was not needed as far as I could tell --- at least, I applied the same diff to my local codebase (modulo any typos I made) and it worked.

Reply:
Great, thanks!